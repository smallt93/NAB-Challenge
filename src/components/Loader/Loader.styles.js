import styled, { css, keyframes } from 'styled-components';

const spin = keyframes`
  0% { -webkit-transform: rotate(0deg); }
  100% { -webkit-transform: rotate(360deg); }
`

export const LoaderWrapper = styled.div`
  border-radius: 50%;
  border: 2px solid ${({ theme }) => theme.colors.white};
  border-top: 2px solid ${({ theme }) => theme.colors.grey};
  border-bottom: 2px solid ${({ theme }) => theme.colors.grey};
  animation: ${spin} 1s linear infinite;
  position: absolute;
  right: 10px;
  z-index: 1000;
  width: 1.3rem;
  height: 1.3rem;

  ${({ theme, size }) => size === 'large' && css`
    border: 4px solid ${theme.colors.white};
    border-top: 4px solid ${theme.colors.grey};
    border-bottom: 4px solid ${theme.colors.grey};
    width: 2.5rem;
    height: 2.5rem;
    margin-bottom: 1em;
    position: unset;
  `}
`;