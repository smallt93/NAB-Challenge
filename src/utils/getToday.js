const weekDay = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];

export const getDayOfWeek = (date) => {
  const getDay = String(date.getDate()).padStart(2, '0');
  return `${weekDay[date.getDay()]} ${getDay}`;
};