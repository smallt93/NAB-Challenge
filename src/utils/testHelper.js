import React from 'react';
import { mount, shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from 'stylesheets/defaultTheme.styles';


export function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}

export function shallowWithTheme(child) {
  return shallow(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}
