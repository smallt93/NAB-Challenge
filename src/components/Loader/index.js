import React from 'react';
import { LoaderWrapper } from './Loader.styles';

const Loader = ({ size = '' }) => {
  return <LoaderWrapper size={size} />;
};

export default Loader;
