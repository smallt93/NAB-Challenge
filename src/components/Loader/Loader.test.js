import React from 'react';
import { mountWithTheme as mount } from 'utils/testHelper';
import Loader from 'components/Loader';

describe('render Loader component', () => {
  let Component;

  const mockProps = {
    size: '',
  }
  
  beforeEach(() => {
    Component = mount(<Loader {...mockProps} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render', () => {
    expect(Component.length).toBe(1);
  });

  it('should match snapshot', () => {
    expect(Component).toMatchSnapshot();
  });
});
