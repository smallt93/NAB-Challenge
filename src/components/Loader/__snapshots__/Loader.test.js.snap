// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Loader component should match snapshot 1`] = `ReactWrapper {}`;
